README.md

```diff
@@ -197,6 +198,20 @@ go_library(name, srcs, deps, data)
         <p>List of files needed by this rule at runtime.</p>
       </td>
     </tr>
+    <tr>
```

Comment:
I think you can keep the rule API simpler.

* I didn't document `cgo_object` by intention because it was an implementation detail to propagate dependency to native c compiler objects.  It is better to keep this configuration hidden from users if possible.
* Similarly, it is better to avoid having a new rule for go assembler for users to keep consistency to mental model of go tool and `cc_library`.
  * We defined a new rule `cgo_library` because there was no clear way for Bazel to distinguish cgo-enabled sources from pure-go sources at runtime. On the other hand, in this case, you can distinguish assembler inputs from go sources by file extension.

So, we should be able to do something like the following if you call `emit_go_asm_action` from `go_library_impl`.
```bzl
go_library(
  name = "go_default_library",
  srcs = ["asm_go.go", "add.s", "sub.s"],
  library = ":foo",
)

cgo_library(
  name = "foo",
  srcs = ["a.go", "b.c", "b.h", "c.cc", "c.h"],
)
```